src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts

```diff
@@ -102,38 +102,46 @@ describe('ExerciseDetailsStudentActionsComponent', () => {
         jest.restoreAllMocks();
     });
 
-    it('should not show the buttons "Team" and "Start exercise" for a team exercise when not assigned to a team yet', fakeAsync(() => {
-        comp.exercise = teamExerciseWithoutTeamAssigned;
+    it.each([ExerciseType.MODELING, ExerciseType.FILE_UPLOAD, ExerciseType.PROGRAMMING, ExerciseType.TEXT])(
+        'should not show the buttons "Team" and "Start exercise" for a team exercise when not assigned to a team yet',
+        fakeAsync((exerciseType: ExerciseType) => {
+            comp.exercise = { ...teamExerciseWithoutTeamAssigned, type: exerciseType };
 
-        fixture.detectChanges();
-        tick();
-
-        const viewTeamButton = fixture.debugElement.query(By.css('.view-team'));
-        expect(viewTeamButton).toBeNull();
+            fixture.detectChanges();
+            tick();
 
-        const startExerciseButton = fixture.debugElement.query(By.css('.start-exercise'));
-        expect(startExerciseButton).toBeNull();
-    }));
+            const viewTeamButton = fixture.debugElement.query(By.css('.view-team'));
+            expect(viewTeamButton).toBeNull();
 
-    it('should show the button "Team" for a team exercise for a student to view his team when assigned to a team', fakeAsync(() => {
-        comp.exercise = teamExerciseWithTeamAssigned;
+            const startExerciseButton = fixture.debugElement.query(By.css('.start-exercise'));
+            expect(startExerciseButton).toBeNull();
+        }),
+    );
 
-        fixture.detectChanges();
-        tick();
+    it.each([ExerciseType.TEXT, ExerciseType.MODELING, ExerciseType.FILE_UPLOAD, ExerciseType.PROGRAMMING])(
+        'should show the button "Team" for a team exercise for a student to view his team when assigned to a team',
+        fakeAsync((exerciseType: ExerciseType) => {
+            comp.exercise = { ...teamExerciseWithTeamAssigned, type: exerciseType };
+            fixture.detectChanges();
+            tick();
 
-        const viewTeamButton = fixture.debugElement.query(By.css('.view-team'));
-        expect(viewTeamButton).not.toBeNull();
-    }));
+            const viewTeamButton = fixture.debugElement.query(By.css('.view-team'));
+            expect(viewTeamButton).not.toBeNull();
+        }),
+    );
 
-    it('should show the button "Start exercise" for a team exercise when assigned to a team', fakeAsync(() => {
-        comp.exercise = teamExerciseWithTeamAssigned;
+    it.each([ExerciseType.PROGRAMMING, ExerciseType.TEXT, ExerciseType.FILE_UPLOAD, ExerciseType.MODELING])(
```

Comment:
Shouldn't we be able to combine this test with the one above as we did with the other two tests?